src/main/java/org/openrewrite/java/dependencies/UpgradeDependencyVersion.java

```diff
@@ -95,7 +95,12 @@ public String getDescription() {
 
     @Override
     public List<Recipe> getRecipeList() {
-        if (upgradeGradleDependencyVersion == null && upgradeMavenDependencyVersion == null) {
+        if (upgradeGradleDependencyVersion == null ||
```

Comment:
I suggest we add a comment here, so that nobody in the future thinks it would make sense to remove this code which looks like it shouldn't be necessary in the first place. Or do we have a test (in this repo) which would fail?

Reply:
Exactly, this is what we discussed yesterday face to face. Do we need to add a comment and add a test that at least mimics how this recipe is invoked.

Reply:
test added. Adding a comment there too